### ISSUE TITLE
applySchemaEnhancer should receive block props

### ISSUE THREAD

**Issue by tiberiuichim** [plone/volto#4156]
It's useful to receive the block props, as that would allow "mixing in" external data, that doesn't belong to either formData or the schema.

https://github.com/plone/volto/blob/cf11e2ef4e3801559be92872e52afff46e990e23/src/helpers/Extensions/withBlockSchemaEnhancer.js#L260

**Comment by yashnerkar** [comment]
can I work on this?

**Comment by tiberiuichim** [comment]
@yashnerkar Yes! Be aware that Volto follows the Plone developer requirements (you need to sign an agreement), see plone.org for details).

Also, https://github.com/plone/volto/issues/3250#issuecomment-1382736643

**Comment by yashnerkar** [comment]
Yess!

**Comment by tiberiuichim** [comment]
@yashnerkar yes, correct.

**Comment by yashnerkar** [comment]
sorry but I didnt get the proper Idea of how am I passing the props through the applySchemaEnhancer. Can you explain a bit?
